### PR TITLE
Update guidance on RAIL

### PR DESCRIPTION
**Pull request by addyosmani**
**What's changed, or what was fixed?**

This is an update to our RAIL guidance including affordances for our newer loading metrics. It also introduces a more explicit call out for goals and guidelines for each area.

The goals and guidelines call outs are something the Chrome Speed Metrics team felt would make for a more clear framing of RAIL.

**Fixes:** #5464 with more details on what changed

**Target Live Date:** December 15th

- [ X] I have run `gulp test` locally and all tests pass.
- [ X] I have added the appropriate `type-something` label.

**R:** @petele @Meggin @kaycebasques 

Notes: This is my last day in the office for this year. If anyone would like to take on directly editing/addressing further changes before the new year, I'm more than happy for them to do so :)

### CODE REVIEW

**Comment by addyosmani** [comment]
Does anyone have feedback on this one? :)

**Comment by kaycebasques** [comment]
I'll check it out, Addy

**Comment by addyosmani** [comment]
Thanks, Kayce! :)

File: src/content/en/fundamentals/performance/rail.md [hunk 1]
```diff
@@ -2,32 +2,40 @@ project_path: /web/fundamentals/_project.yaml
...
+By laying out a structure for thinking about performance, RAIL enables
+designers and developers to reliably target the highest-impact work.
 
-![RAIL performance model](images/rail.png)
+Every web app has these four distinct aspects to its life cycle, and performance 
+fits into them in different ways:
 
+![RAIL performance model](images/rail.png)
 
 ### TL;DR {: .hide-from-toc }
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: Change to "Summary"

Comment (kaycebasques) [reply src/content/en/fundamentals/performance/rail.md]: Or "Overview"

File: src/content/en/fundamentals/performance/rail.md [hunk 2]
```diff
@@ -2,32 +2,40 @@ project_path: /web/fundamentals/_project.yaml
...
 
-- Focus on the user; the end goal isn't to make your site perform fast on any specific device, it's to ultimately make users happy.
+- Focus on the user; the end goal isn't to make your site perform fast on any
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: s/;/./

File: src/content/en/fundamentals/performance/rail.md [hunk 3]
```diff
@@ -2,32 +2,40 @@ project_path: /web/fundamentals/_project.yaml
...
 
-- Focus on the user; the end goal isn't to make your site perform fast on any specific device, it's to ultimately make users happy.
+- Focus on the user; the end goal isn't to make your site perform fast on any 
+specific device, it's to ultimately make users happy.
 - Respond to users immediately; acknowledge user input in under 100ms.
 - When animating or scrolling, produce a frame in under 10ms.
 - Maximize main thread idle time.
-- Keep users engaged; deliver interactive content in under 1000ms.
-
+- Keep users engaged; deliver interactive content in under 5s on slow 3G on median mobile phones
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: Rephrase

File: src/content/en/fundamentals/performance/rail.md [hunk 4]
```diff
@@ -67,12 +75,34 @@ Understand how users perceive performance delays:
...
+## Response: respond in under 50ms {: #response }
 
-You have 100ms to respond to user input before they notice a lag.
-This applies to most inputs, such as clicking buttons, toggling form
-controls, or starting animations. This does not apply to touch drags or
-scrolls.
+**Guideline**: You have 100ms to perform work related to transitioning.
+
+**Goal**:
+We want to transition in under this time and suggest you have 50ms to
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: @addyosmani general writing feedback: our style guide recommends avoiding "we" except in certain circumstances: https://developers.google.com/style/person

When an article represents our team's perspective, it's sometimes appropriate to use "we" (but we need to explicitly define who "we" is). In many cases it's more appropriate to use "you", because it clearly demonstrates who needs to take the action

File: src/content/en/fundamentals/performance/rail.md [hunk 5]
```diff
@@ -67,12 +75,34 @@ Understand how users perceive performance delays:
...
+either. Due to packet loss and network variance, their **effective connection
+type** could be significantly slower.
+
+## Goals and guidelines
+
+RAIL offers a set of performance **guidelines** which map closely to set of 
+**goals** regarding user-facing performance metrics. We believe that our
+guidelines can help you achieve the goals.
+
+* **Goals** aim to be timeless
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: Interesting way to organize the thoughts!

File: src/content/en/fundamentals/performance/rail.md [hunk 6]
```diff
@@ -67,12 +75,34 @@ Understand how users perceive performance delays:
   </tbody>
 </table>
 
-## Response: respond in under 100ms {: #response }
+The perception of performance delays can vary based on mobile hardware
+and network conditions. While loading an experience in 1000ms is plausible on
+a powerful desktop machine over cable, on mobile, ensuring users can load and 
+interact with content in under 5000s is a more [realistic](http://chimera.labs.oreilly.com/books/1230000000545/ch08.html#MOBILE_NETWORK_LATENCY_OVERHEAD) target under slow 3G.
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: @addyosmani two points (no action needed on your part)

1. this URL redirects to https://hpbn.co/optimizing-for-mobile-networks/ is this the same content as what you inteded?
2. Our style guide recommends descriptive link texts so readers have a better idea of what they're opening https://developers.google.com/style/link-text

**Comment by kaycebasques** [comment]
(just a note that I started updating the doc and it's incomplete, so no one else should push commits to this branch or merge this PR)

**Comment by PaulKinlan** [comment]
Added do not merge tag to make it clearer. Let us know when you are ready to land.

**Comment by kaycebasques** [comment]
Thanks @PaulKinlan, don't know why I didn't think of that

Working on this now!

**Comment by kaycebasques** [comment]
Biggest updates:

* I really like the "goals/guidelines" organization, but when I went through the doc, the sections for R, A, I, L weren't following the definitions for "goals" and "guidelines" that we set out. "Goals" are timeless, "guidelines" are practical tips designed for today's conditions, so they may change over time. So I updated each section to try to better match these definitions.
* Added a lot of content to the tools sections.

Considering the size of the changes, I think we've got to hand this back to @addyosmani to review (when he gets back). I've also got a few Qs that we need to resolve (adding them to lines of code after this comment)

All in all I think his new organization and my polish will be a big improvement to the guidance

File: src/content/en/fundamentals/performance/rail.md [hunk 7]
```diff
@@ -1,171 +1,350 @@
...
+For example, loading an experience in 1000ms is plausible on a powerful desktop machine over a
+fast Wi-Fi connection, so users have grown accustomed to a 1000ms loading experience. But for
+mobile devices over slow 3G connections, loading in 5000ms is a more realistic goal, so mobile
+users are generally more patient.
 
-You have 100ms to respond to user input before they notice a lag.
-This applies to most inputs, such as clicking buttons, toggling form
-controls, or starting animations. This does not apply to touch drags or
-scrolls.
+## Response: respond in under 50ms {: #response }
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: @addyosmani can you review this section, I'm not sure if I captured it correctly. Transition needs to complete by 100ms? And the app needs to provide feedback within 50ms?

File: src/content/en/fundamentals/performance/rail.md [hunk 8]
```diff
@@ -1,171 +1,350 @@
...
-Animations aren't just fancy UI effects. For example, scrolling and touch
-drags are types of animations.
+**Goal**: Produce frames as fast as the user's hardware can display them. Users notice when
+animation frame rates vary.
+
+**Guidelines**:
+
+* Produce each frame in an animation in 10ms or less. Technically, the maximum budget for
+  each frame is 16ms (1000ms / 60 frames per second ≈ 16ms), but browsers need about 6ms to
+  render each frame, hence the guideline of 10ms per frame.
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: Isn't this tied to human perception? I.e. we perceive something as smooth if it's 60FPS. So shouldn't this be the goal?

File: src/content/en/fundamentals/performance/rail.md [hunk 9]
```diff
@@ -1,171 +1,350 @@
...
+**Guidelines**:
 
-For actions that take longer than 500ms to complete, always provide feedback.
+* Respond to user input within 50ms, or else the connection between action and
+  reaction is broken. This applies to most inputs, such as clicking buttons, toggling form
+  controls, or starting animations. This does not apply to touch drags or scrolls.
+* Though it may sound counterintuitive, it's not always the right call to respond to user
+  input immediately. You can use this 100ms window to do other expensive work. But be
+  careful not to block the user. If possible, do work in the background.
+* For actions that take longer than 500ms to complete, always provide feedback.
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: Is this supposed to be 50ms?

File: src/content/en/fundamentals/performance/rail.md [hunk 10]
```diff
@@ -1,171 +1,350 @@
...
+See [The Need For Mobile Speed: How Mobile Latency Impacts Publisher Revenue][NEED4SPEED].
+
+[NEED4SPEED]: https://www.doubleclickbygoogle.com/articles/mobile-speed-matters/
+
+**Goals**: 
+
+* For first loads, load the page and be [interactive][tti] in 5 seconds or less on mid-range
+  mobile devices with slow 3G connections. See [Can You Afford It? Real-World Web Performance
+  Budgets][Budgets].
+* For subsequent loads, load the page in under 2 seconds.
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: Aren't these "goals" actually guidelines? The metric for a "fast load" is relative to the status quo. 1s load time for mobile devices is unrealistic. But if status quo conditions change over time, then 3s (for example) may become realistic. Therefore we'll need to change the "goal", therefore it's not "timeless"

I'm just wondering if there's a way to phrase this that makes it truly timeless guidance

Comment (kaycebasques) [reply src/content/en/fundamentals/performance/rail.md]: Like maybe "be aware of the typical load performance that your users experience, based on their devices and network connections, then aim for a fast loading experience relative to those constraints"

File: src/content/en/fundamentals/performance/rail.md [hunk 11]
```diff
@@ -1,171 +1,350 @@
...
+The **Performance** panel is the main place to analyze your RAIL metrics. See [Get Started With
+Analyzing Runtime Performance][cdt-runtime] to get familiar with the **Performance** panel UI.
+The workflow and UI for analyzing load performance is mostly the same, the only difference is how
+you start and stop the recording. See [Record load performance][cdt-load].
+
+[cdt-runtime]: /web/tools/chrome-devtools/evaluate-performance/
+[cdt-load]: /web/tools/chrome-devtools/evaluate-performance/reference#record-load
+
+The following DevTools features are especially relevant:
+
```
Comment (kaycebasques) [inline src/content/en/fundamentals/performance/rail.md]: Admittedly I could probably organize this section a little better, but I think it's a good start.

**Comment by petele** [comment]
@addyosmani - can you take a look please?

File: src/content/en/fundamentals/performance/rail.md (hunk 9 above)
Comment (addyosmani) [reply src/content/en/fundamentals/performance/rail.md]: You're absolutely right - this should be 50ms.

File: src/content/en/fundamentals/performance/rail.md (hunk 8 above)
Comment (addyosmani) [reply src/content/en/fundamentals/performance/rail.md]: Thinking back through this, you're right to call out that we don't tie this back to human perception right now. I think optimizing for that and ensuring smoothness is pretty valid for the goal. Did you have additional edits you'd like to make here?

File: src/content/en/fundamentals/performance/rail.md (hunk 10 above)
Comment (addyosmani) [reply src/content/en/fundamentals/performance/rail.md]: Perhaps, "Optimize for fast loading performance relative to the devices and network conditions your users use to access your site"? I agree that for the guidance to be truly timeless it shouldn't be tied to a specific point-in-time constraint like TTI in 5s (we may change this in a year or two). 

What I'd like us for the guideline of TTI in <5s on median range mobile devices over slow 3G to be something captured lower down in the doc

File: src/content/en/fundamentals/performance/rail.md (hunk 11 above)
Comment (addyosmani) [reply src/content/en/fundamentals/performance/rail.md]: I'm a fan of this idea of explicitly calling out features that are relevant to RAIL here across our tooling better. I vote for filing an issue to rethink how we can organize this section better but moving ahead with what we've got here for now.

File: src/content/en/fundamentals/performance/rail.md (hunk 8 above)
Comment (kaycebasques) [reply src/content/en/fundamentals/performance/rail.md]: I'm going to move "Produce each frame in 10ms..." up to the Goals list

**Comment by petele** [comment]
@kaycebasques - when you merge, would you please choose the "Squash and Merge" option. Thanks!!

**Comment by petele** [comment]
Oh, and don't forget to delete the branch.

**Comment by kaycebasques** [comment]
Got it, thanks Pete. I was wondering how we kept the commit history clean
